### ISSUE TITLE
Prefer Scala's Duration string conversion to Typesafe Config's implementation.

### ISSUE THREAD

**Issue by leifwickland** [pureconfig/pureconfig#59]
I tried using pureconfig in a Spark 1.x app and discovered that I couldn't parse a `FiniteDuration`. Spark 1.x includes Typesafe Config 1.2.1 , which doesn't include the `Config.getDuration` method. (Pureconfig uses 1.3.1.) At runtime I get an error about that method not existing. Of course the Spark version of typesafe supersedes whatever I ship.

The specific error message is:
```
Exception in thread "main" java.lang.NoSuchMethodError: com.typesafe.config.Config.getDuration(Ljava/lang/String;)Ljava/time/Duration;
        at pureconfig.DurationConvert$$anonfun$from$2.apply(DurationConvert.scala:20)
...
```

Well, in the process of working around that (which kudos to everybody here that I can easily enough work around that) I realized that Scala's `Duration` includes a `def apply(String): Duration` that is (almost) a superset of `SimpleConfig.parseDuration`.

See
https://github.com/typesafehub/config/blob/master/config/src/main/java/com/typesafe/config/impl/SimpleConfig.java#L622
vs
https://github.com/scala/scala/blob/v2.11.8/src/library/scala/concurrent/duration/Duration.scala#L78

The "almost" is that for abbreviating microseconds Scala used a proper _mu_ while typesafe config used a _u_, i.e. "µs" vs "us".

When I originally implemented pureconfig's converter for `Duration` I didn't realize that Scala had its own implementation of something similar. I was shooting for compatibility with Typesafe config.

Pureconfig's `Duration` converter is one of its most complex. It also includes the "config to string" converter I wrote, which Scala's duration includes.

**Proposal**:
- Use Scala's `Duration.apply` instead of `Config.getDuration`
- Maintain compatibility with typesafe config by translating "us" to "µs".
- Discard most of pureconfig's duration converter. It could be replaced by something like ConfigConvert.fromNonEmptyString(Duration.apply(_)). (With the addition of the unit converter mentioned above.)

@melrief @jcazevedo, if you guys are marginally supportive, I'll send a PR for you to consider further.

An interesting side effect of this change is that it'd help #5, support for Java 7. Pureconfig could offer converters for `Duration` and `FiniteDuration` without typesafe config 1.3.x.

**Comment by jcazevedo** [comment]
We can use the `long getDuration(String path, TimeUnit unit)` method instead (which is available at least since Typesafe Config [v1.2.0](https://github.com/typesafehub/config/blob/v1.2.0/config/src/main/java/com/typesafe/config/Config.java#L635)), but I'm not against adding a `Duration` conversion that doesn't depend on Typesafe Config's `getDuration` methods directly, as we already provide helpers that don't use Typesafe Config's counterparts (`Boolean`, `Double`, `Float`, `Int`, `Long` and `Short` are all converted from `String`).

**Comment by melrief** [comment]
I agree, I would just try to be as much compatible with the Typesafe Config way to do it as possible because I think users expects this.
